src/java/org/apache/cassandra/metrics/KeyspaceMetrics.java

```diff
@@ -275,6 +279,10 @@ public KeyspaceMetrics(final Keyspace ks)
         rowIndexSizeAborts = createKeyspaceMeter("RowIndexSizeAborts");
         rowIndexSize = createKeyspaceHistogram("RowIndexSize", false);
 
+        tooManyIndexReadSSTablesWarnings = createKeyspaceMeter("TooManyIndexReadSSTablesWarnings");
+        tooManyIndexReadSSTablesAborts = createKeyspaceMeter("TooManyIndexReadSSTablesAborts");
+        tooManyIndexReadSSTables = createKeyspaceHistogram("TooManyIndexReadSSTables", false);
```

Comment:
I noticed we already have this histogram in `TableQueryMetrics.PerQueryMetrics.sstablesHit` BUT (and @mike-tr-adamson can correct me here) it looks like it's only updated for top-k queries? We should probably just make sure update that for all queries and just remove this `TooManyIndexReadSSTables` histogram.

Reply:
`TableQueryMetrics.PerQueryMetrics.sstablesHit` is used on all index search paths. It is updated from `QueryContext.sstablesHit` and that is updated in the `IndexSearchResultIterator`.

Reply:
I stand corrected. Thanks Mike. Yeah, feels like we can just remove the `tooManyIndexReadSSTables` histogram then.